### DISCUSSION TITLE
courses/computer-structure/%ED%94%84%EB%A1%9C%EC%84%B8%EC%84%9C/02_%EC%82%B0%EC%88%A0%EB%85%BC%EB%A6%AC%EC%9E%A5%EC%B9%98

### DISCUSSION THREAD

**Discussion by giscus** [crwdu/lms-contents-js101#66] [Q&A]
# courses/computer-structure/%ED%94%84%EB%A1%9C%EC%84%B8%EC%84%9C/02_%EC%82%B0%EC%88%A0%EB%85%BC%EB%A6%AC%EC%9E%A5%EC%B9%98

https://lms.vanillacoding.co/courses/computer-structure/%ED%94%84%EB%A1%9C%EC%84%B8%EC%84%9C/02_%EC%82%B0%EC%88%A0%EB%85%BC%EB%A6%AC%EC%9E%A5%EC%B9%98

**Comment by junesung1004** [comment]
퀴즈
-----------------------------------------------------------------------------------------------
1. ALU에서 사용하는 데이터 단위는 무엇인가요? bit 입니다.

2. 숫자 6 + 9를 계산할 때 ALU는 어떤 과정을 거칠까요?
ㄴ 먼저 숫자 6을 8비트 2진수로 변환하면 00000110 로 표현할 수 있습니다.
ㄴ 그다음 숫자 9를 8비트 2진수로 변환하면 00001001 로 표현할 수 있습니다.
ㄴ     00000110
ㄴ +  00001001      2개의 2진수로 변환한 두 식을 더하면
ㄴ     00001111 값이 나온다.
ㄴ   8 + 4 + 2 + 1 = 15 라는 값이 나온다.

3. 음수는 비트로 어떻게 처리될까요?
-> 공부해서 돌아오겠습니다..

복습 문제
-----------------------------------------------------------------------------------------------
1. 숫자 8을 왼쪽으로 2비트 쉬프트(Shift Left)하면 어떤 값이 나올까요?
ㄴ 숫자 8을 2진수 8비트로 표현하면 0 0 0 0 1 0 0 0 이 나온다.
ㄴ 왼쪽으로 쉬프트 연산을 하면 2의 배수씩 증가한다. 
ㄴ 이를 2진수로 표현하면 0 0 0 1 0 0 0 0 로 표현할 수 있다(1비트가 증가)
ㄴ 위 1비트 왼쪽 시프트연산을 적용한 값을 계산하면 16이 나온다
ㄴ 위 과정에서 1비트 왼쪽 시프트 연산을 적용하면 0 0 1 0 0 0 0 0
ㄴ 즉 계산을 한 결과 값은 32가 나온다.

* 쉬프트연산 과정에서 주의할 점은 오버플로 현상이 일어날 수 있다.
 ■ 1 0 0 0 0 0 0 1 
 위 값에서 왼쪽 시프트 연산을 진행하면 -> 0 0 0 0 0 0 1 0 이라는 값이 적용되어 
 위 로직의 중요한 시점이었다면 프로그램은 치명적인 에러를 발생했을 것이다.


2. 10 - 7의 계산 과정을 비트로 표현해 보세요.
ㄴ 10(00001010) - 7(00000111)

 (00001010)
-(00000111)
---------------
(00000011)

즉 3의 값이 나온다. (2 + 1)

**Reply by Ken123777** [reply]
음수를 처리하는 방법 기대하고 있겠습니다 🙇🏻‍♂️
